dht.go

```diff
@@ -702,3 +703,11 @@ func (dht *IpfsDHT) newContextWithLocalTags(ctx context.Context, extraTags ...ta
 	) // ignoring error as it is unrelated to the actual function of this code.
 	return ctx
 }
+
+func (dht *IpfsDHT) maybeAddAddrs(p peer.ID, addrs []ma.Multiaddr, ttl time.Duration) {
+	// Don't add addresses for self or our connected peers. We have better ones.
+	if p == dht.self || dht.host.Network().Connectedness(p) == network.Connected {
```

Comment:
not adding addresses because the peer is connected seems counter intuitive. when are the peer's addresses added in that case?

Reply:
Because we don't want someone else to tell us the _wrong_ addresses for the peer, overwriting the _good_ ones in our peerstore.